pkg/arvo/sys/hoon.hoon

```diff
@@ -6440,43 +6512,45 @@
   ^-  toon
   ?.  ?=({$2 *} ton)  ton
   :-  %2
-  :: =.  p.ton  (moop p.ton)
-  =+  yel=(lent p.ton)
-  =.  p.ton
-    ?.  (gth yel 1.024)  p.ton
+  =+  yel=(lent trace.ton)
+  =.  trace.ton
+    ?.  (gth yel 1.024)  trace.ton
     %+  weld
-      (scag 512 p.ton)
+      (scag 512 trace.ton)
     ^-  (list {@ta *})
-    :_  (slag (sub yel 512) p.ton)
+    :_  (slag (sub yel 512) trace.ton)
     :-  %lose
     %+  rap  3
     "[skipped {(scow %ud (sub yel 1.024))} frames]"
   |-  ^-  (list tank)
-  ?~  p.ton  ~
-  =+  rep=$(p.ton t.p.ton)
-  ?+    -.i.p.ton  rep
-      $hunk  [(tank +.i.p.ton) rep]
-      $lose  [[%leaf (rip 3 (@ +.i.p.ton))] rep]
-      $hand  [[%leaf (scow %p (mug +.i.p.ton))] rep]
-      $mean  :_  rep
-             ?@  +.i.p.ton  [%leaf (rip 3 (@ +.i.p.ton))]
-             =+  mac=(mack +.i.p.ton +<.i.p.ton)
-             ?~(mac [%leaf "####"] (tank u.mac))
-      $spot  :_  rep
-             =+  sot=(spot +.i.p.ton)
+  ?~  trace.ton  ~
+  =+  rep=$(trace.ton t.trace.ton)
+  =*  dat  +.i.trace.ton
+  ?+    -.i.trace.ton  rep
+      $hunk  =/  sof=(unit tank)  ((soft tank) dat)
+             ?~  sof  rep
+             [u.sof rep]
+      $lose  ?^  dat  rep
+             :_  rep
+             leaf+(rip 3 dat)
+      $hand  :_  rep
+             leaf+(scow %p (mug dat))
+      $mean  ?@  dat  [leaf+(rip 3 dat) rep]
+             =/  mac  (mack dat -.dat)
+             ?~  mac  [leaf+"####" rep]
+             =/  sof  ((soft tank) u.mac)
+             ?~  sof  rep
+             [u.sof rep]
+      $spot  =/  sof=(unit spot)  ((soft spot) dat)
+             ?~  sof  rep
+             :_  rep
```

Comment:
I dislike have so many virtualization contexts in an inner loop, and re-entrant calls into them (`+soft` eventually calls `+mook`). Catching crashes in the stack trace production is very different from catching crashes in arbitrary code -- it's basically for shaking out bugs in the compiler or runtime. Stack trace production can already be pretty slow, and this is likely to make it slower. At the same time, it's more elegant -- and probably more correct -- to not crash here.

Regardless of the virtualization, it'd be nice to refactor the non-standard recursion here. This would easily translate to `+turn` or `+murn`, and the result should be simpler.